include/node.tcc

```diff
@@ -169,6 +172,14 @@ void mpm::Node<Tdim, Tdof, Tnphases>::update_mass_pressure(
     node_mutex_.lock();
     pressure_(phase) += mass_pressure / mass_(phase);
     node_mutex_.unlock();
+
+    if (pressure_constraints_.find(phase) != pressure_constraints_.end()) {
+      const double scalar =
+          (pressure_function_.find(phase) != pressure_function_.end())
+              ? pressure_function_[phase]->value(step * dt)
+              : 1.0;
+      this->pressure_(phase) = scalar * pressure_constraints_[phase];
+    }
```

Comment:
Woudln't it be better to make this in another function instead of appending it in update_mass_pressure? First is because you are doing it everytime this function is called from particles. Actually you can do it just once after every `pressure_` is finally updated.

Reply:
Please see the new design